File: src/test/java/com/koddy/server/common/mock/stub/StubOAuthUriGenerator.java
package com.koddy.server.common.mock.stub;

import com.koddy.server.auth.application.adapter.OAuthUriGenerator;
import com.koddy.server.auth.domain.model.oauth.OAuthProvider;

public class StubOAuthUriGenerator implements OAuthUriGenerator {
    public static final String BASE_URL = "https://localhost:3000/%s?redirectUri=%s";

    @Override
    public String generate(final OAuthProvider provider, final String redirectUri) {
        return String.format(BASE_URL, provider.getValue(), redirectUri);
    }
}

File: src/test/java/com/koddy/server/common/mock/stub/StubOAuthUriGenerator.kt
package com.koddy.server.common.mock.stub

import com.koddy.server.auth.application.adapter.OAuthUriGenerator
import com.koddy.server.auth.domain.model.oauth.OAuthProvider

open class StubOAuthUriGenerator : OAuthUriGenerator {
    override fun generate(
        provider: OAuthProvider,
        redirectUri: String,
    ): String = String.format(BASE_URL, provider.value, redirectUri)

    companion object {
        const val BASE_URL: String = "https://localhost:3000/login/%s?redirectUri=%s"
    }
}
